src/storage/src/table/streaming_table/state_table.rs

```diff
@@ -723,8 +726,70 @@ impl<S: StateStore> StateTable<S> {
         &'a self,
         pk_prefix: &'a Row,
     ) -> StorageResult<RowStream<'a, S>> {
-        let (mem_table_iter, storage_iter_stream) =
-            self.iter_inner(pk_prefix, self.epoch()).await?;
+        let (mem_table_iter, storage_iter_stream) = self
+            .iter_with_pk_prefix_inner(pk_prefix, self.epoch())
+            .await?;
+
+        let storage_iter = storage_iter_stream.into_stream();
+        Ok(
+            StateTableRowIter::new(mem_table_iter, storage_iter, self.row_deserializer.clone())
+                .into_stream()
+                .map(Self::get_second),
+        )
+    }
+
+    /// This function scans rows from the relational table with specific `pk_prefix`.
+    pub async fn iter_with_pk_range<'a>(
+        &'a self,
+        pk_range: &'a (Bound<Row>, Bound<Row>),
+        // Optional vnode that returns an iterator only over the given range under that vnode.
+        // For now, we require this parameter, and will panic. In the future, when `None`, we can
+        // iterate over each vnode that the `StateTable` owns.
+        vnode: Option<u8>,
```

Comment:
```suggestion
        vnode: u8,
```
I think there will be no stream executor scanning data in multiple vnode at all. so maybe we can force the streaming executor give the vnode.